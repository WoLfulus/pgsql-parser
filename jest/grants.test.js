const parser = require('../src');
import { cleanTree, cleanLines } from './utils';
const dosql = require('fs').readFileSync(__dirname + '/fixtures/grants.sql').toString();

describe('grants', () => {
  it('parse and deparse', () => {
    const tree = parser.parse(dosql);
    expect(tree).toMatchSnapshot();
    const sql = parser.deparse(tree.query);
    expect(cleanLines(sql)).toMatchSnapshot();
    expect(cleanTree(parser.parse(sql))).toEqual(cleanTree(tree));
  });
});
