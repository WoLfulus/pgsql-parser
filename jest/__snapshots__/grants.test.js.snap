// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grants parse and deparse 1`] = `
Object {
  "query": Array [
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "GrantStmt": Object {
            "behavior": 0,
            "grantees": Array [
              Object {
                "RoleSpec": Object {
                  "location": 31,
                  "rolename": "administrator",
                  "roletype": 0,
                },
              },
            ],
            "is_grant": true,
            "objects": Array [
              Object {
                "String": Object {
                  "str": "users",
                },
              },
            ],
            "objtype": 10,
            "privileges": Array [
              Object {
                "AccessPriv": Object {
                  "priv_name": "usage",
                },
              },
            ],
            "targtype": 0,
          },
        },
        "stmt_len": 44,
      },
    },
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "GrantStmt": Object {
            "behavior": 0,
            "grantees": Array [
              Object {
                "RoleSpec": Object {
                  "location": 93,
                  "rolename": "anonymous",
                  "roletype": 0,
                },
              },
            ],
            "is_grant": true,
            "objects": Array [
              Object {
                "ObjectWithArgs": Object {
                  "args_unspecified": true,
                  "objname": Array [
                    Object {
                      "String": Object {
                        "str": "auth",
                      },
                    },
                    Object {
                      "String": Object {
                        "str": "authenticate",
                      },
                    },
                  ],
                },
              },
            ],
            "objtype": 7,
            "privileges": Array [
              Object {
                "AccessPriv": Object {
                  "priv_name": "execute",
                },
              },
            ],
            "targtype": 0,
          },
        },
        "stmt_len": 57,
        "stmt_location": 45,
      },
    },
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "GrantStmt": Object {
            "behavior": 0,
            "grantees": Array [
              Object {
                "RoleSpec": Object {
                  "location": 161,
                  "rolename": "administrator",
                  "roletype": 0,
                },
              },
            ],
            "is_grant": true,
            "objects": Array [
              Object {
                "RangeVar": Object {
                  "inh": true,
                  "location": 147,
                  "relname": "token",
                  "relpersistence": "p",
                  "schemaname": "auth",
                },
              },
            ],
            "objtype": 1,
            "privileges": Array [
              Object {
                "AccessPriv": Object {
                  "priv_name": "select",
                },
              },
              Object {
                "AccessPriv": Object {
                  "priv_name": "insert",
                },
              },
              Object {
                "AccessPriv": Object {
                  "priv_name": "update",
                },
              },
              Object {
                "AccessPriv": Object {
                  "priv_name": "delete",
                },
              },
            ],
            "targtype": 0,
          },
        },
        "stmt_len": 71,
        "stmt_location": 103,
      },
    },
  ],
  "stderr": "",
}
`;

exports[`grants parse and deparse 2`] = `
"GRANT USAGE ON SCHEMA users TO \\"administrator\\";
GRANT EXECUTE ON FUNCTION \\"auth\\".\\"authenticate\\" TO \\"anonymous\\";
GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE \\"auth\\" . \\"token\\" TO \\"administrator\\";"
`;
